src/module.c

```diff
@@ -11052,16 +11060,21 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data) {
             } else if (eid == REDISMODULE_EVENT_CONFIG) {
                 moduledata = data;
             } else if (eid == REDISMODULE_EVENT_KEY) {
-                moduledata = data;
-                RedisModuleKeyInfoV1 *ki = data;
-                serverAssert(ki->dbnum != -1);
-                selectDb(ctx.client, ki->dbnum);
+                KeyInfo *info = data;
+                selectDb(ctx.client, info->dbnum);
+                moduleInitKey(&key, &ctx, info->key, info->value, info->mode);
+                RedisModuleKeyInfoV1 ki = {info->version, &key};
```

Comment:
i think it's a bit off to place this struct declaration on the stack inside the `if` and use that memory outside.
i know it works, but still, let's move it to the top of the function